lib/runtime/wasmer/imports.go

```diff
@@ -234,44 +383,172 @@ func ext_crypto_secp256k1_ecdsa_recover_compressed_version_1(context unsafe.Poin
+		return 0
+	}
+
+	for i, key := range keys {
+		copy(memory[ptr+uint32(i*32):ptr+uint32((i+1)*32)], key.Encode())
+	}
+
+	ret := pointerAndSizeToInt64(int32(ptr), int32(len(keys)*32))
+	return C.int64_t(ret)
 }
 
 //export ext_crypto_sr25519_sign_version_1
-func ext_crypto_sr25519_sign_version_1(context unsafe.Pointer, a, z C.int32_t, y C.int64_t) C.int64_t {
+func ext_crypto_sr25519_sign_version_1(context unsafe.Pointer, keyTypeID, key C.int32_t, msg C.int64_t) C.int64_t {
 	logger.Trace("[ext_crypto_sr25519_sign_version_1] executing...")
-	logger.Warn("[ext_crypto_sr25519_sign_version_1] unimplemented")
-	return 0
+	instanceContext := wasm.IntoInstanceContext(context)
+	runtimeCtx := instanceContext.Data().(*runtime.Context)
+	memory := instanceContext.Memory().Data()
+
+	pubKey, err := sr25519.NewPublicKey(memory[key : key+32])
+	if err != nil {
+		logger.Error("[ext_crypto_sr25519_sign_version_1] failed to get public key", "error", err)
+		return 0
+	}
+
+	signingKey := runtimeCtx.Keystore.GetKeypair(pubKey)
+	if signingKey == nil {
+		logger.Error("[ext_crypto_sr25519_sign_version_1] could not find key in keystore", pubKey)
+		return 0
+	}
+
+	msgData := asMemorySlice(instanceContext, msg)
+	sig, err := signingKey.Sign(msgData)
+	fmt.Println("sig", sig)
+	if err != nil {
+		logger.Error("[ext_crypto_sr25519_sign_version_1] could not sign message")
+		return 0
+	}
+
+	ret, err := toWasmMemoryOptional(instanceContext, sig)
+	if err != nil {
+		logger.Error("[ext_crypto_sr25519_sign_version_1] failed to allocate memory", err)
+		return 0
+	}
+
+	return C.int64_t(ret)
 }
 
 //export ext_crypto_sr25519_verify_version_1
-func ext_crypto_sr25519_verify_version_1(context unsafe.Pointer, a C.int32_t, z C.int64_t, y C.int32_t) C.int32_t {
+func ext_crypto_sr25519_verify_version_1(context unsafe.Pointer, sig C.int32_t, msg C.int64_t, key C.int32_t) C.int32_t {
 	logger.Trace("[ext_crypto_sr25519_verify_version_1] executing...")
-	logger.Warn("[ext_crypto_sr25519_verify_version_1] unimplemented")
-	return 0
+
+	instanceContext := wasm.IntoInstanceContext(context)
+	memory := instanceContext.Memory().Data()
+	signVerify := instanceContext.Data().(*runtime.Context).SigVerifier
+
+	newMsg := asMemorySlice(instanceContext, msg)
+	newSig := memory[sig : sig+64]
+
+	pub, err := sr25519.NewPublicKey(memory[key : key+32])
+	if err != nil {
+		return 0
+	}
+
+	if signVerify.IsStarted() {
+		signature := runtime.Signature{
+			PubKey:    pub.Encode(),
+			Sign:      newSig,
+			Msg:       newMsg,
+			KyeTypeID: crypto.Sr25519Type,
+		}
+		signVerify.Add(&signature)
+		return 0
+	}
+
+	if ok, err := pub.Verify(newMsg, newSig); err != nil || !ok {
+		return 0
+	}
+	return 1
 }
 
 //export ext_crypto_sr25519_verify_version_2
-func ext_crypto_sr25519_verify_version_2(context unsafe.Pointer, a C.int32_t, z C.int64_t, y C.int32_t) C.int32_t {
+func ext_crypto_sr25519_verify_version_2(context unsafe.Pointer, sig C.int32_t, msg C.int64_t, key C.int32_t) C.int32_t {
```

Comment:
it seems like version 1 and version 2 of this func use different versions of the underlying crypto library, will confirm the difference